docs/source/architecture/staking_contracts.rst

```diff
@@ -0,0 +1,155 @@
+single Worker. Each token holder can deposit any amount of NU into the pool and will be entitled to the pro-rata
+share of the pool and rewards without needing to maintain and run their own Worker. Token holders will pay a
+percentage of NU staking rewards received to the owner of the Worker for running a node.
+
+There is the added benefit of reducing overall Worker gas costs by combining multiple Stakers, each of whom would
+have needed to a Worker otherwise, into one large Staker that uses a single Worker that performs work on
+behalf of the staking pool.
+
+
+Contract Roles
+^^^^^^^^^^^^^^
+
+The pooling contract has several roles:
+
+* *Owner* - controls the staking functions including all parameters of staking e.g. ``restake``, ``winddown``, etc.
+
+    .. important::
+
+      It is recommended to use a multisig, DAO or other decentralized governance mechanism.
+
+* *Worker Owner* - the owner of the Worker that runs on behalf of the staking pool; only this address can
+  withdraw the worker's collected fee
+* *Delegators* - the NU holders who deposit NU into the pooling contract
+* *Administrator (Optional)* - oversees contract upgrades which allow modification to all contract logic
+  and behaviour; this role only applies when the contract is upgradeable and the OpenZeppelin's ``Proxy`` contract
+  is utilized
+
+    .. warning::
+
+      Be cautious of who is bestowed the Administrator role because even if the pooling contract was audited,
+      the Administrator can modify/upgrade the contract code after deployment. **It is recommended to use a
+      multisig, DAO or other decentralized governance mechanism.**
+
+
+Worker Fees
+^^^^^^^^^^^
+The *Worker Owner* receives a percentage of NU staking rewards as a fee for running a Worker on behalf of the
+staking pool. This percentage is configured during contract deployment. The implication here is that if the pool
+does not generate any rewards then the *Worker Owner* will not receive any fee.
+
+
+Contract Lifecycle
+^^^^^^^^^^^^^^^^^^
+
+* The *Owner* deploys the contract and initializes it by specifying the Worker fee percentage and the *Worker Owner*
+  address.
+* Once deployed, deposits are enabled by default to start accepting deposits from *Delegators*.
+* At any point before the *Owner* creates a stake, a Delegator can exit the pool and recoup their original deposit via
+  the ``withdrawAll`` function.
+* After the intended deposits have been received, the *Owner* specifies staking parameters to create a stake: size,
+  duration, restaking, winddown etc., and bonds the stake to the Worker address.
+* Once the *Owner* creates a stake, the ability to deposit into the pool is automatically disabled to prevent any
+  new deposits. Disabling deposits ensures that there is clear proportional ownership of the pool and its received
+  rewards. This is a much simpler model for determining proportional ownership than allowing deposits after
```

Comment:
Can we explain a bit more in detail how pool ownership is used to calculate returns?

Reply:
✔️